spring-boot-project/spring-boot/src/test/java/org/springframework/boot/info/ProcessInfoTests.java

```diff
@@ -54,4 +59,28 @@ void memoryInfoIsAvailable() {
 		assertThat(nonHeapUsageInfo.getMax()).isEqualTo(-1);
 	}
 
+	@Test
+	void virtualThreadsInfoIsNullWhenMXBeanIsNotAccessible() {
+		if (!isVirtualThreadMXBeanAvailable()) {
+			ProcessInfo processInfo = new ProcessInfo();
+
+			VirtualThreadsInfo virtualThreadsInfo = processInfo.getVirtualThreads();
+
+			assertThat(virtualThreadsInfo).isNull();
+		}
+	}
+
+	private boolean isVirtualThreadMXBeanAvailable() {
```

Comment:
I think ClassUtils.isPresent("jdk.management.VirtualThreadSchedulerMXBean") could be enough here.